pkg/types/field.go

```diff
@@ -401,7 +401,7 @@ func (f *Field) AddToResource(g *Builder, r *resource, typeNames *TypeNames, add
 // an earlier step, so they cannot be included as well. Plus probably they
 // should also not change for Create and Update steps.
 func (f *Field) isInit() bool {
-	return !f.Identifier && f.Reference == nil && (f.TFTag != "-" || f.Injected)
```

Comment:
Why do we delete `f.Injected` condition here? For injected fields (for defining associative lists in the context of SSA), we need the TF tag to be `-` so that they are ignored during TF JSON serialization but we still need them to be available in the CRD API. So, I suspect, we need to preserve the condition of `f.Injected` here.

Reply:
Thanks @ulucinar for catching this. After rebasing, I tried to solve the conflicts and it seems this removal is not an intentional change. I put this again.